libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+            _rangefinder_dist_m = temp * 0.1f;
+            break;
+        }
+
+        case GimbalSubCmd::GimbalSubCmd_GimbalSn:{
+            // gimbal model, length is over 14 bytes and less than 18 bytes
+            if(_parsed_msg.msg.data_len >= 14){
+                _got_model_name = true;
+                if(_parsed_msg.msg.data_len > 18){
+                    _parsed_msg.msg.data_len = 18;
+                }
+                
+                strncpy((char *)_model_sn, (const char *)_parsed_msg.msg.data, _parsed_msg.msg.data_len);
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s SN: %s", send_text_prefix, (const char*)_model_sn);
+            }
+            break;
+        }
+
+        default:
+            break;           
+        }
+        break;
+    }
+
+    case GimbalCmd::MasCmd_Device: {
+        switch ((DeviceSubCmd)_parsed_msg.msg.sub_cmd) {
+        case DeviceSubCmd::Firmware_Info: {
+            if (_parsed_msg.msg.data_len >= 10) { 
+                    _got_firmware_version = true;
+
+                // print both app&boot version and compiling time in UTC+8
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s AppVersion: %d.%d.%d %02d-%02d-%02d %02d:%02d", send_text_prefix, 
+                        (unsigned char)_parsed_msg.msg.data[0],  
+                        (unsigned char)_parsed_msg.msg.data[1],  
+                        (unsigned char)_parsed_msg.msg.data[2],  
+                        (unsigned char)_parsed_msg.msg.data[3],  
+                        (unsigned char)_parsed_msg.msg.data[4],  
+                        (unsigned char)_parsed_msg.msg.data[5], 
+                        (unsigned char)_parsed_msg.msg.data[6],
+                        (unsigned char)_parsed_msg.msg.data[7]); 
+                            
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s BootVersion: %d.%d.%d %02d-%02d-%02d %02d:%02d", send_text_prefix,                                    
+                        (unsigned char)_parsed_msg.msg.data[8],
+                        (unsigned char)_parsed_msg.msg.data[9],
+                        (unsigned char)_parsed_msg.msg.data[10],
+                        (unsigned char)_parsed_msg.msg.data[11],
+                        (unsigned char)_parsed_msg.msg.data[12],
+                        (unsigned char)_parsed_msg.msg.data[13],
+                        (unsigned char)_parsed_msg.msg.data[14], 
+                        (unsigned char)_parsed_msg.msg.data[15]);
+                    break;
+            }
+            break;
+        }
+        default:
+            break;
+        }
+        break;
+    }
+
+    case GimbalCmd::MasCmd_Load:{
+
+        break;
+    }
+
+    default:
+        break;
+    }
+}
+
+// send packet to gimbal.  total_len includes everything including the header '0x5A' and crc
+// returns true on success, false if outgoing serial buffer is full or other errors
+bool AP_Mount_Tusuav::send_packet(struct FrameType frame)
```

Comment:
```suggestion
bool AP_Mount_Tusuav::send_packet(const struct FrameType &frame)
```

... we've already got a send buffer on the stack here, don't need to update the frame's total length, really.

Reply:
Thank you for the suggestion. We’ve updated the function to use `const struct FrameType &frame` .

Reply:
Thank you for the suggestion. We’ve updated the function to use const struct FrameType &frame.